Sources/NIOCore/ActorExecutor.swift

```diff
@@ -0,0 +1,82 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if swift(>=5.9)
```

Comment:
Just circling back here. If we want there is a way to create a custom serial executor without warnings on 5.9 and blow like this

```
#if swift(<5.9)
final class MyExecutor: SerialExecutor {
    func enqueue(_ job: UnownedJob) {
        job._runSynchronously(on: self.asUnownedSerialExecutor())
    }

    func asUnownedSerialExecutor() -> UnownedSerialExecutor {
        UnownedSerialExecutor(ordinary: self)
    }
}
#else
@available(macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0, *)
final class MyExecutor: SerialExecutor {
    @available(macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0, *)
    func enqueue(_ job: __owned ExecutorJob) {
      job.runSynchronously(on: asUnownedSerialExecutor())
    }

    func asUnownedSerialExecutor() -> UnownedSerialExecutor {
        UnownedSerialExecutor(ordinary: self)
    }
}
#endif
```

Reply:
Thanks Franz. It's a good suggestion but I'm inclined not to do it: using both `_runSynchronously` _and_ `__owned` makes me really quite nervous about the stability of that code. @weissi are you happy with having this API on 5.9 only?

Reply:
I'm happy with having APIs that are available everywhere from a certain Swift version. But it should be `#if compiler` not `#if swift`